File: app/src/main/java/com/wilinz/yuetingmusic/data/model/FavoriteSong.java
package com.wilinz.yuetingmusic.data.model;

import android.net.Uri;

import org.litepal.annotation.Column;
import org.litepal.crud.LitePalSupport;

public class FavoriteSong extends LitePalSupport {
    public int id;
    public User user;
//    @Column(unique = true)
    public Song song;
    public long song_id;
    public long timestamp;
    @Column(nullable = false, unique = true, index = true)
    public String uniqueId;

    @Override
    public String toString() {
        return "FavoriteSong{" +
                "id=" + id +
                ", user=" + user +
                ", song=" + song +
                ", song_id=" + song_id +
                ", timestamp=" + timestamp +
                '}';
    }
}

File: app/src/main/java/com/wilinz/yuetingmusic/data/model/FavoriteSong.kt
package com.wilinz.yuetingmusic.data.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import org.litepal.annotation.Column
import org.litepal.crud.LitePalSupport

@Parcelize
data class FavoriteSong(
    var id: Int = 0,
    var user: User? = null,
//    @Column(unique = true)
    var song: Song? = null,
    var song_id: Long = 0,
    var timestamp: Long = 0,
    @Column(nullable = false, unique = true, index = true)
    var uniqueId: String? = null
) : LitePalSupport(), Parcelable
